pulsar/internal/connection.go

```diff
@@ -424,7 +422,24 @@ func (c *connection) handleResponse(requestID uint64, response *pb.BaseCommand)
 
 	delete(c.pendingReqs, requestID)
 	c.mapMutex.RUnlock()
-	request.callback(response)
+	request.callback(response, nil)
+}
+
+func (c *connection) handleResponseError(serverError *pb.CommandError) {
+	requestID := serverError.GetRequestId()
+	c.mapMutex.RLock()
```

Comment:
Should we get a Lock() here since we might be deleting and entry from the map?

Reply:
Uhm, this locking was introduced in a7c1a045. 

I think that was not a correct fix. All the internal methods should be accessed only from the connection goroutine. If that was not the case, we should rather fix that aspect.